jekyll/_cci2/manage-contexts-with-config-policies.adoc

```diff
@@ -0,0 +1,523 @@
+
+. Replace `<context-1>` with the name of context you want to all the use of. You can add multiple contexts as an array, or stick with one for now.
+** To find a list of your contexts, select **Organization settings** from the sidebar in the CircleCI web app and then select **Contexts** from the menu.
+
+[#push-up-your-policy-bundle-3]
+=== 3. Push up your policy bundle
+
+You can now push your new policy to your organization for it to take affect. You have two options, either push the policy manually using the CLI from your local environment, or push your changes to your config policy repo if you are managing policies via your VCS as shown in the xref:create-and-manage-config-policies#manage-policies-with-your-vcs[Manage policies with your VCS guide].
+
+[tab.reserve.manual]
+--
+Create and upload the policy bundle using CircleCI CLI:
+
+[source,shell]
+----
+circleci policy push ./config-policies –owner-id <your-organization-ID>
+----
+
+If the upload was successful, you will see something like the following:
+
+[source,shell]
+----
+{
+  “Created”: [“reserved_contexts”]
+}
+----
+--
+
+[tab.reserve.push_to_vcs]
+--
+If you have set up your config policies repository with the sample configuration shown in the xref:create-and-manage-config-policies#manage-policies-with-your-vcs[Manage policies with your VCS guide], push your changes to the `main` branch of your config policies repository, and head to the CircleCI web app to see your policy pipeline run.
+
+You can also push to a development branch, in which case you will get a diff of your policy bundle when you push your changes, rather than your new policy being pushed to your CircleCI organization. This is useful when developing your policies.
+--
+
+NOTE: If you would like to write tests for your policy, check out the xref:test-config-policies#[Test config policies] guide.
+
+[#conclusion-3]
+=== Conclusion
+Once your have pushed your new `reserve_contexts.rego` policy, if an attempt to trigger a pipeline is made, in which the specified project has access to contexts in the allow-list, but the project is not included in the reserve list configured in your policy, the pipeline will fail to trigger, and developers will be notified on the dashboard.
+
+[#define-the-contexts-reserved-by-branch]
+== Define the contexts reserved by a branch
+
+You may want to restrict which contexts (and therefore secrets) are available depending on which branch is being built. Using this paradigm, you can manage your application environment in one repository, and lock down the use of secrets to individual branches, for example splitting up production secrets and development secrets. This allows you to ensure that production secrets cannot be accessed byt a build on a development branch. Use the xref:config-policy-reference#contexts-reserved-by-branches[`contexts_reserved_by_branches`] helper to define a policy for this use case.
```

Comment:
```suggestion
You may want to restrict which contexts (and therefore secrets) are available depending on which branch is being built. Using this paradigm, you can manage your application environment in one repository, and lock down the use of secrets to individual branches, for example, splitting up production secrets and development secrets. This ensures that production secrets cannot be accessed by a build on a development branch. Use the xref:config-policy-reference#contexts-reserved-by-branches[`contexts_reserved_by_branches`] helper to define a policy for this use case.
```
Any chance of replacing "paradigm" with another word?

Reply:
Yeah I was wondering about that one too 😅 . How about:

> Using branch-based restrictions, you can manage your application environment ...